components/popupSearch/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React, {useEffect, useState} from "react";
+import {
+  CloseIconStyles,
+  Input,
+  InputContainer,
+  Popup,
+  PopupBackground,
+  SearchListContainer
+} from "./popupSearch.styled";
+import {SearchItem} from "../search/SearchItem";
+import CloseIcon from '../../assets/icons/closeIcon.svg'
+import {SearchErrorMessage} from "../search/SearchErrorMessage";
+import {HeaderProps} from "../../interfaces/header";
+import {PopupSearchItemType} from "../../interfaces/popupSearch";
+
+export const PopupSearch: React.FC<HeaderProps> = ({setIsPopup, isPopup}) => {
+  const [searchText, setSearchText] = useState('')
+  const [searchList, setSearchList] = useState<PopupSearchItemType[] | []>([])
+
+  const fetchApi = async (searchText: string) => {
```

Comment:
you have 3 places calling fetch. So it makes sense to create helper for them.
Let say tomorrow we gonna move to axios instead of fetch. then you would have to replace it in every place you used fetch

Reply:
pls check if I did it right

Reply:
almost.
why have you skipped error handling part?
 catch (e) {
      setSearchList([])
      console.log('e', e)
    }

Reply:
sorry, found it now. But it is better to move error handling into getData function, as any request might have failed. e.g. server is not responding